core/src/replay_stage.rs

```diff
@@ -3877,7 +3925,25 @@ impl ReplayStage {
                 if bank.is_frozen() && tower.is_slot_confirmed(*slot, voted_stakes, total_stake) {
                     info!("validator fork confirmed {} {}ms", *slot, duration);
                     datapoint_info!("validator-confirmation", ("duration_ms", duration, i64));
-                    confirmed_forks.push((*slot, bank.hash()));
+                    confirmed_forks.push(ConfirmedSlot::new_optimistic_confirmed_slot(
```

Comment:
open to ideas about naming, but also I think we should just fix the plumbing and make it roll up while we're looking at this.

The plumbing rn seems to be a mess:
- `cluster_info_vote_listener` tracks gossip votes but also receives replay votes through crossbeam from `replay_stage`
- `cluster_info_vote_listener` tracks gossip votes for propagated check in `vote_tracker`
- `replay_stage` also tracks replay votes in `vote_tracker` and rolls up in `fork_stats` in `compute_bank_stats`
- `cluster_info_vote_listener` also separately sends verified gossip votes to `replay_stage` through crossbeam
- `replay_stage` tracks `DuplicateConfirmed` through `fork_stats` but also gets notified through crossbeam from `cluster_info_vote_listener` 

There's a lot of back and forth propagation of votes and stake info in multiple places which seems inefficient. My vote would be to just share 1 vote tracker/fork stats between `cluster_info_vote_listener` and `replay_stage` to avoid the extra work. This would also let us rollup replay votes *and* include gossip votes that would get us over the key threshold.




If we don't want to do a big refactor I think there's a simpler solution for rollup:
https://github.com/solana-labs/solana/blob/d58db6e2559783ee51c10857d703909b600bee61/ledger/src/blockstore_processor.rs#L171-L175
When sending replayed votes from `replay_stage` to `cluster_info_vote_listener`, send the slot hashes as well. 

Then
https://github.com/solana-labs/solana/blob/d58db6e2559783ee51c10857d703909b600bee61/core/src/cluster_info_vote_listener.rs#L677-L680
If we have `!is_gossip` we don't have to filter only on the last slot here, we can track the whole tower since we have the accompanying hash in slot hashes.

Reply:
to clarify this^ is out of scope for this pr and should not be a blocker for merging. we can discuss in https://github.com/solana-labs/solana/issues/34279.

I agree that optimistically confirmed is an overloaded name and am happy to rename if you have a suggestion.

Reply:
optimistically confirmed actually can't be safely rolled up, it has to be all votes on the same slot

Reply:
I think just called it `supermajority_confirmed`?